pyaedt/modules/report_templates.py

```diff
@@ -2444,40 +2431,569 @@ def _context(self):
+            X points units. The default is ``"ns"``.
+        yunits : str, optional
+            Y points units. The default is ``"mV"``.
+        enable_limits : bool, optional
+            Whether to enable the upper and lower limits. The default is ``False``.
+        upper_limits float, optional
+            Upper limit if limits are enabled. The default is ``500``.
+        lower_limits str, optional
+            Lower limit if limits are enabled. The default is ``-500``.
+        color : tuple, optional
+            Mask in (R, G, B) color. The default is ``(0, 255, 0)``.
+            Each color value must be an integer in a range from 0 to 255.
+        xoffset : str, optional
+            Mask time offset with units. The default is ``"0ns"``.
+        yoffset : str, optional
+            Mask value offset with units. The default is ``"0V"``.
+        transparency : float, optional
+            Mask transparency. The default is ``0.3``.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+        """
+        if "quantity_type" in dir(self) and self.report_type == "Rectangular Contour Plot":
+            props = [
+                "NAME:AllTabs",
+                ["NAME:Mask", ["NAME:PropServers", "{}: Plot {}".format(self.plot_name, self.traces[0].name)]],
+            ]
+        else:
+            props = [
+                "NAME:AllTabs",
+                ["NAME:Mask", ["NAME:PropServers", "{}:EyeDisplayTypeProperty".format(self.plot_name)]],
+            ]
+        arg = [
+            "NAME:Mask",
+            "Version:=",
+            1,
+            "ShowLimits:=",
+            enable_limits,
+            "UpperLimit:=",
+            upper_limit if upper_limit else 1,
+            "LowerLimit:=",
+            lower_limit if lower_limit else 0,
+            "XUnits:=",
+            xunits,
+            "YUnits:=",
+            yunits,
+        ]
+        mask_points = ["NAME:MaskPoints"]
+        for point in points:
+            mask_points.append(point[0])
+            mask_points.append(point[1])
+        arg.append(mask_points)
+        args = ["NAME:ChangedProps", arg]
+        if not ("quantity_type" in dir(self) and self.report_type == "Rectangular Contour Plot"):
+            args.append(["NAME:Mask Fill Color", "R:=", color[0], "G:=", color[1], "B:=", color[2]])
+            args.append(["NAME:X Offset", "Value:=", xoffset])
+            args.append(["NAME:Y Offset", "Value:=", yoffset])
+            args.append(["NAME:Mask Trans", "Transparency:=", transparency])
+        props[1].append(args)
+        self._post.oreportsetup.ChangeProperty(props)
+
+        return True
+
+    @pyaedt_function_handler()
+    def rectangular_plot(self, value=True):
+        """Enable or disable the rectangular plot on the chart.
+
+        Parameters
+        ----------
+        value : bool
+            Whether to enable the rectangular plot. The default is ``True``. When
```

Comment:
```suggestion
            Whether to enable the rectangular plot. The default is ``True``. If
```
We try to consistenly explain the non-default behavior with "If X, ...".